internal/pkg/completion/completion.go

```diff
@@ -0,0 +1,81 @@
+package completion
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+func CompletionBash(out io.Writer, cmd *cobra.Command) error {
+	err := cmd.Root().GenBashCompletion(out)
+
+	// The default binary name downloaded from the Releases page is dtm-{os}-amd64
+	// solve the problem that autocompletion fails when the name of the binary is not dtm
+	if binary := filepath.Base(os.Args[0]); binary != "dtm" {
+		renamedBinary := `
+# the user renamed the dtm binary
+if [[ $(type -t compopt) = "builtin" ]]; then
+    complete -o default -F __start_dtm %[1]s
+else
+    complete -o default -o nospace -F __start_dtm %[1]s
+fi
+`
+		fmt.Fprintf(out, renamedBinary, binary)
+	}
+
+	return err
+}
+
+func CompletionZsh(out io.Writer, cmd *cobra.Command) error {
+	err := cmd.Root().GenZshCompletionNoDesc(out)
+
+	// The default binary name downloaded from the Releases page is dtm-{os}-amd64
+	// solve the problem that autocompletion fails when the name of the binary is not dtm
+	if binary := filepath.Base(os.Args[0]); binary != "dtm" {
+		renamedBinary := `
+# the user renamed the dtm binary
+compdef _dtm %[1]s
+`
+		fmt.Fprintf(out, renamedBinary, binary)
+	}
+
+	fmt.Fprintf(out, "compdef _dtm dtm")
+
+	return err
+}
+
+func BashExample(binary string) string {
+	return fmt.Sprintf(`Load is completions in the current shell session:
+# source <(%s completion bash)
+
+Load completions for every new session:
+(in Linux)# %s completion bash > /etc/bash_completion.d/dtm
+(in MacOS)# %s completion bash > /usr/local/etc/bash_completion.d/dtm`, binary, binary, binary)
```

Comment:
`/usr/local/` can be changed to `$(brew --prefix)`, because only Mac computers with Intel processors is `/usr/local/`, but M1 is `/opt/homebrew`